ttnn/cpp/ttnn/operations/data_movement/slice/slice.hpp

```diff
@@ -16,38 +16,44 @@ struct SliceOperation {
         const ttnn::Tensor& input_tensor,
         tt::tt_metal::Shape output_tensor_start,
         tt::tt_metal::Shape output_tensor_end,
-        const std::optional<MemoryConfig>& memory_config_arg); 
+        const std::optional<tt::tt_metal::Shape> step,
```

Comment:
Any way it can be non-optional with a proper default value?

Reply:
Yes, with all ones it can be done. Any particular reason?

Probably best to switch it to an int vector as well to make it ONNX-like.